scripts/dev/host_deploy_ipks

```diff
@@ -0,0 +1,236 @@
+        """
+        fqdn = getfqdn()
+        print("=== Copy+Paste to setup opkg config ===")
+        print("cat >/etc/opkg/dev.conf <<-EOF")
+        for feed_name in self.feed_names:
+            print(f"src  {feed_name}  http://{fqdn}:{port}/{feed_name}  [trusted=yes]")
+        print("EOF")
+        print("===")
+
+
+def find_ipk_feeds(search_root: os.PathLike=os.getcwd(), skip_empty: bool=False) -> list[Path]:
+    """Search a path (`search_root`) for 'Package' index files, optionally
+    filter out those which are empty, and return the results as a list.
+
+    Args:
+        search_root: The base path from which to start the search.
+        skip_empty: If True, do not include indexes which contain no 'Package:'
+            entries in the search results.
+
+    Returns:
+        A list of paths to the found indexes, relative to the `search_root`.
+    """
+    logger.info(f"Searching {search_root} for IPK feeds.")
+    ipk_feeds = []
+    for index in glob("**/Packages", root_dir=search_root, recursive=True):
+        index_full_path = Path(search_root) / index
+        feed_path = index_full_path.parent
+        logger.info(f"Found IPK feed: {feed_path}")
+
+        if skip_empty:
+            with open(index_full_path, "r") as fp_index:
+                if not re.search(r"^Package: ", fp_index.read(), flags=re.MULTILINE):
+                    logger.warning(f"Skipping empty index: {index_full_path}")
+                    continue
+
+        ipk_feeds.append(feed_path)
+    return ipk_feeds
+
+def parse_args(argv: list[str]) -> Namespace:
+    """Parse a CLI argument sequence and set namespace variables relevant to
+    this module's `main()` method.
+    
+    Args:
+        argv: The CLI argument sequence to parse.
+
+    Returns: A populated namespace of arguments, for use by the main() method.
+    """
+    parser = ArgumentParser(description=__doc__, allow_abbrev=False)
+    parser.add_argument("--verbose", "-v", action="count", default=0,
+        help="Increase the verbosity of script output. Specify once for verbose output, specify twice for debug.")
+    parser.add_argument("--version", "-V", action="store_true",
+        help="Print his script's version information and exit.")
+    
+    parser.add_argument("-b", "--bind", nargs="?", default=HTTP_DEFAULT_HOST,
+        help="The host address to which the feed server should bind. (Default=%(default)s)")
+    parser.add_argument("-e", "--host-empty", action="store_true",
+        help="If asserted, host IPK feeds that contain no packages.")
+    parser.add_argument("-p", "--port", type=int, nargs="?", default=HTTP_DEFAULT_PORT,
+        help="The port number over which the feed server will host. (Default=%(default)s)")
+    parser.add_argument("--suppress-opkg-conf", action="store_true",
+        help="If enabled, do not print an opkg conf script.")
+    
+    parser.add_argument("root", type=Path, nargs="?", default=os.getcwd(),
```

Comment:
[Optional] It'd be nice if the script can autolocate this as location of feeds is fixed wrt location of the script.

Reply:
That would mean that calling the script with no arguments would start hosting content on the network. I don't want that to be the case as a safety for if someone accidentally tab-completes or dmenu's the script.

I could add a `--auto` argument or something to try and auto-locate the deploy IPKs. But at that point, you may as well instead pass the path yourself.

So I'm leaning toward not doing this.